src/python_testing/mdns_discovery/mdns_async_service_info.py

```diff
@@ -0,0 +1,273 @@
+        port: int = _MDNS_PORT,
+        record_type: DNSRecordType = None
+    ) -> bool:
+        """Returns true if the service could be discovered on the
+        network, and updates this object with details discovered.
+
+        This method will be run in the event loop.
+
+        Passing addr and port is optional, and will default to the
+        mDNS multicast address and port. This is useful for directing
+        requests to a specific host that may be able to respond across
+        subnets.
+        """
+        if not self._zc.started:
+            await self._zc.async_wait_for_start()
+
+        now = current_time_millis()
+
+        if TYPE_CHECKING:
+            assert self._zc.loop is not None
+
+        first_request = True
+        delay = self._get_initial_delay()
+        next_ = now
+        last = now + timeout
+        try:
+            self.async_add_listener(self, None)
+            while not self._is_complete:
+                if last <= now:
+                    return False
+                if next_ <= now:
+                    this_question_type = question_type or QU_QUESTION if first_request else QM_QUESTION
+                    out: DNSOutgoing = self._generate_request_query(self._zc, now, this_question_type, record_type)
+                    first_request = False
+                    if out.questions:
+                        # All questions may have been suppressed
+                        # by the question history, so nothing to send,
+                        # but keep waiting for answers in case another
+                        # client on the network is asking the same
+                        # question or they have not arrived yet.
+                        self._zc.async_send(out, addr, port)
+                    next_ = now + delay
+                    next_ += self._get_random_delay()
+                    if this_question_type is QM_QUESTION and delay < _DUPLICATE_QUESTION_INTERVAL:
+                        # If we just asked a QM question, we need to
+                        # wait at least the duplicate question interval
+                        # before asking another QM question otherwise
+                        # its likely to be suppressed by the question
+                        # history of the remote responder.
+                        delay = _DUPLICATE_QUESTION_INTERVAL
+
+                await self.async_wait(min(next_, last) - now, self._zc.loop)
+                now = current_time_millis()
+        finally:
+            self._zc.async_remove_listener(self)
+
+        return True
+
+    def async_add_listener(
+        self, listener: RecordUpdateListener, question: Optional[Union[DNSQuestion, List[DNSQuestion]]]
+    ) -> None:
+        """Adds a listener for a given question.  The listener will have
+        its update_record method called when information is available to
+        answer the question(s).
+
+        This function is not threadsafe and must be called in the eventloop.
```

Comment:
Is this the chip stack event loop or something else?

Reply:
This code is part of a method from a zeroconf class that has been overridden, though this particular code isn't the part being personalized, this executes as is in the zeroconf library overridden (in this case) or not.